tactics/tactics.ml

```diff
@@ -2863,17 +2863,22 @@ let generalize_dep ?(with_let=false) c =
       | _ -> tothin
   in
   let cl' = it_mkNamedProd_or_LetIn (pf_concl gl) to_quantify in
-  let body =
-    if with_let then
-      match EConstr.kind sigma c with
-      | Var id -> id |> (fun id -> pf_get_hyp id gl) |> NamedDecl.get_value
-      | _ -> None
-    else None
+  let is_var, body = match EConstr.kind sigma c with
+  | Var id ->
+    let body = id |> (fun id -> pf_get_hyp id gl) |> NamedDecl.get_value in
```

Comment:
What is this strange pattern? Can’t you simply write the following?

~~~
let body = pf_get_hyp id gl |> NamedDecl.get_value in
~~~

Reply:
I didn't write that, that's code I moved around. But indeed this is strange.

Reply:
Rewritten.